eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -153,7 +159,13 @@ class Peer(nodeParams: NodeParams, remoteNodeId: PublicKey, authenticator: Actor
     case Event(Terminated(actor), d: InitializingData) if d.channels.exists(_._2 == actor) =>
       val h = d.channels.filter(_._2 == actor).keys
       log.info(s"channel closed: channelId=${h.mkString("/")}")
-      stay using d.copy(channels = d.channels -- h)
+      val channels1 = d.channels -- h
+      if (channels1.isEmpty) {
+        // we have no existing channels, we can forget about this peer
+        stop(FSM.Normal)
+      } else {
+        stay using d.copy(channels = d.channels -- h)
```

Comment:
To avoid redoing the diff
```suggestion
        stay using d.copy(channels = channels1)
```